provider/pkg/schema/gen_tags.go

```diff
@@ -29,24 +31,62 @@ const (
 	TagsStyleKeyValueArrayWithAlternateType TagsStyle = "keyValueArrayWithAlternateType"
 )
 
-func GetTagsProperty(originalSpec *jsschema.Schema) (string, bool) {
-	tagging, ok := originalSpec.Root().Extras["tagging"]
-	if !ok {
-		return "", false
-	}
-	asInterface, ok := tagging.(map[string]interface{})
-	if !ok {
-		return "", false
+func (ts TagsStyle) IsStringMap() bool {
+	return ts == TagsStyleStringMap
+}
+
+func (ts TagsStyle) IsKeyValueArray() bool {
+	return strings.HasPrefix(string(ts), "keyValueArray")
+}
+
+func (ctx *context) ApplyTagsTransformation(propertySpec *pschema.PropertySpec, spec *jsschema.Schema, tagsStyle TagsStyle) {
+	switch tagsStyle {
+	case TagsStyleUntyped:
+	case TagsStyleStringMap:
+		// Nothing to do
+	case TagsStyleKeyValueArray:
+		// Swap referenced type to shared definition and remove custom type.
+		propertySpec.TypeSpec.Items.Ref = "#/types/" + globalTagToken
+	case TagsStyleKeyValueArrayCreateOnly:
+		// Swap referenced type to shared definition and remove custom type.
+		propertySpec.TypeSpec.Items.Ref = "#/types/" + globalCreateOnlyTagToken
+	case TagsStyleKeyValueArrayWithExtraProperties:
+		// Keep custom type
+	case TagsStyleKeyValueArrayWithAlternateType:
+		// Keep custom type
+	default: // Unknown
+		ctx.reports.UnexpectedTagsShapes[ctx.resourceToken] = spec
 	}
-	tagProperty, ok := asInterface["tagProperty"]
-	if !ok {
-		return "", false
+}
+
+func GetTagsProperty(originalSpec *jsschema.Schema) (string, bool) {
+	fromTaggingMetadata, ok := func() (string, bool) {
```

Comment:
What's the reason for using an inline function that's instantly run? Isn't that the same as inlining the function body?